kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java

```diff
@@ -3,41 +3,46 @@
 import com.codeborne.selenide.SelenideElement;
 import io.qameta.allure.Step;
-import org.openqa.selenium.By;
-import org.openqa.selenium.Keys;
 
-import static com.codeborne.selenide.Selenide.$;
+import java.util.Arrays;
+
+import static com.codeborne.selenide.Selenide.$x;
 import static com.codeborne.selenide.Selenide.sleep;
+import static com.provectus.kafka.ui.utilities.WebUtils.clearByKeyboard;
 import static com.provectus.kafka.ui.utilities.WebUtils.clickByJavaScript;
 import static com.provectus.kafka.ui.utilities.screenshots.Screenshooter.log;
 
 public class ConnectorDetails {
-    protected SelenideElement dotMenuBtn = $(By.xpath("//button[@aria-label='Dropdown Toggle']"));
-    protected SelenideElement deleteBtn = $(By.xpath("//li/div[text()='Delete']"));
-    protected SelenideElement confirmBtnMdl = $(By.xpath("//div[@role='dialog']//button[text()='Confirm']"));
-    protected SelenideElement submitBtn = $(By.xpath("//button[@type='submit']"));
-    protected SelenideElement contentTextArea = $("[wrap]");
+
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement dotMenuBtn = $x("//button[@aria-label='Dropdown Toggle']");
+    protected SelenideElement deleteBtn = $x("//li/div[contains(text(),'Delete')]");
+    protected SelenideElement confirmBtnMdl = $x("//div[@role='dialog']//button[contains(text(),'Confirm')]");
+    protected SelenideElement submitBtn = $x("//button[@type='submit']");
+    protected SelenideElement contentTextArea = $x("//textarea[@class='ace_text-input']");
+    protected SelenideElement taskTab = $x("//a[contains(text(),'Tasks')]");
+    protected SelenideElement configTab = $x("//a[contains(text(),'Config')]");
+    protected SelenideElement configField = $x("//div[@id='config']");
 
     @Step
     public ConnectorDetails waitUntilScreenReady() {
-        $(By.xpath("//a[text() ='Tasks']")).shouldBe(Condition.visible);
-        $(By.xpath("//a[text() ='Config']")).shouldBe(Condition.visible);
+        loadingSpinner.shouldBe(Condition.disappear);
+        Arrays.asList(taskTab,configTab).forEach(elementsMenu -> elementsMenu.shouldBe(Condition.visible));
         return this;
     }
 
     @Step()
     public ConnectorDetails openConfigTab() {
-        clickByJavaScript($(By.xpath("//a[text() ='Config']")));
+        clickByJavaScript(configTab);
         return this;
     }
 
     @Step()
     public ConnectorDetails setConfig(String configJson) {
-        $("#config").click();
-        contentTextArea.sendKeys(Keys.LEFT_CONTROL + "a");
-        contentTextArea.setValue("");
-        contentTextArea.setValue(String.valueOf(configJson.toCharArray()));
-        $("#config").click();
+        configField.shouldBe(Condition.enabled).click();
+        clearByKeyboard(contentTextArea);
+        contentTextArea.setValue(configJson);
+        configField.shouldBe(Condition.enabled).click();
         clickByJavaScript(submitBtn);
         sleep(4000);
```

Comment:
can we do smth with this hardcoded wait?

Reply:
sure , deleted

Reply:
And added alertMessage condition